src/include/platform/internal/GenericPlatformManagerImpl_POSIX.cpp

```diff
@@ -132,18 +133,36 @@ CHIP_ERROR GenericPlatformManagerImpl_POSIX<ImplClass>::_StartChipTimer(int64_t
 template <class ImplClass>
 void GenericPlatformManagerImpl_POSIX<ImplClass>::_PostEvent(const ChipDeviceEvent * event)
 {
+    LockEventQueue();
     mChipEventQueue.push(*event); // Thread safe due to ChipStackLock taken by App thread
     SysOnEventSignal(this);       // Trigger wake select on CHIP thread
+    UnlockEventQueue();
```

Comment:
Locking around the usage of the event, it's a deadlock if any event handler posts.

**Potential deadlock**